spec/integration/outputs/elasticsearch/node_spec.rb

```diff
@@ -0,0 +1,35 @@
+require "spec/es_spec_helper"
+require "logstash/outputs/elasticsearch/protocol"
+
+describe "elasticsearch node client" do
+  # Test ElasticSearch Node Client
+  # Reference: http://www.elasticsearch.org/guide/reference/modules/discovery/zen/
+
+  it "should support hosts in both string and array" do
+    # Because we defined *hosts* method in NodeClient as private,
+    # we use *obj.send :method,[args...]* to call method *hosts*
+    client = LogStash::Outputs::Elasticsearch::Protocols::NodeClient.new(:host => "localhost")
+
+    # Node client should support host in string
+    # Case 1: default :host in string
+    insist { client.send :hosts, :host => "host",:port => 9300 } == "host:9300"
+    # Case 2: :port =~ /^\d+_\d+$/
+    insist { client.send :hosts, :host => "host",:port => "9300-9302"} == "host:9300,host:9301,host:9302"
```

Comment:
I would rather break all cases in a different test example as all test different things.